### ISSUE TITLE
Show font information in (PDF) file Properties window

### ISSUE THREAD

**Issue by sahwar** [sumatrapdfreader/sumatrapdf#23]
pdffonts is a handy tool supplied in the Poppler library. It shows the information of all the fonts used in a PDF file: name, type, encoding, embeded or not etc.

Is it possible to use this in SumatraPDF to show font information as a separate tab in the document Properties dialog? It's always very useful to know which fonts and in what encoding are included in a PDF if for example you're trying to recreate it in HTML5+CSS3 for instance by using https://github.com/coolwanglu/pdf2htmlEX.

**Comment by ousia** [comment]
@zeniko, many thanks for the implementation.

**Comment by sudoeric** [comment]
Any update on this? Would be great if it was built into the default version. Evince (Linux) does it just fine and fast as far as I can tell, maybe look into how they've done it?

**Comment by RadGH** [comment]
@sudoeric I found myself in need of this feature today. The CTRL+D twice trick doesn't work in Sumatra 3.1.1  and the one on zeniko.ch seems to be outdated (3.1).

But I found Evince is available for Windows, and the font detection works just fine. Thank you for the suggestion!

I hope this gets added to Sumatra some day.

Here is what Evince looks like, and it was available as soon as I was able to find the menu, no loading bar needed:

![evince_2017-02-21_10-57-14](https://cloud.githubusercontent.com/assets/2008464/23179938/8db9b1ca-f824-11e6-884b-ca8a178fc41c.jpg)

**Comment by andselisk** [comment]
The irony is that since Feb 2018 Evince is [only available for Linux flavors or as a source code](https://wiki.gnome.org/Apps/Evince/Downloads). [They don't build Windows versions anymore]((https://mail.gnome.org/archives/evince-list/2018-February/msg00006.html)) since

> there is no-one that can maintain that. 

TL; DR: It would be really nice if SumatraPDF were able to reveal font info reliably and out of the box.

**Comment by ousia** [comment]
@andselisk, the link to the _Windows_ binary for _Evince_ is https://mail.gnome.org/archives/evince-list/2018-April/msg00002.html.

@zeniko, contains a binary, but it shows no font info. Would it be possible that you contribute the code to the development version of _SumatraPDF_?

**Comment by RadGH** [comment]
I came back to this over a year later and found myself using the Evince version that Ousia pointed out 
(thanks). Still hoping Sumatra adds the feature some day!

**Comment by GitHubRulesOK** [comment]
from the links above
Evince w32-3.26.0-1-installer is about 200 MB or evince-w64-3.26.0-2-installer.exe = 205.64 MB
after a long time it unpacks to over 700MB (that's more than 30,300 files in 1,453 folders) WTF !!

from above SumatraPDF-dev.exe (one single file suited to portable use) is 2.52 MB

Evince should be able to knock the spots of SumatraPDF when it finally loads (admittedly it does have basic annotation etc)
**however**
Sumatra can report Courier New and Georgia for its settings.txt file (try that in Evince and you get File type Text Document (text/plain) is not supported)
for the Bible de Genève, 1564 example downloaded from first link SumatraPDF reports EXACTLY THE SAME 3 variants of EBGaramond as evince does
However SumatraPDF will directly open **URL**  http://dl.fullcirclemagazine.org/issue65_en.pdf from the third  example (can't try that in Evince) and report 9 unique fonts including 5 Scribus Outlines (Fo7SO etc.) whereas Evince needs the file to be downloaded first and reports 30 fonts (4 fonts reported multiple times) and can't name the 5 Scribus Outline Fonts

**Comment by RadGH** [comment]
@GitHubRulesOK I wasn't trying to say that Evince was better or anything. I love Sumatra PDF and prefer it over anything else for reading PDFs. But I get design files from designers who think PDFs are a viable format for web design (it's not) and I need to detect fonts in the PDF to know what specific families, weights and styles are used.

Sumatra to my knowledge can't detect fonts. There was apparently an option to press CTRL+D twice but that doesn't do anything.

Are you saying that Sumatra does have this feature? How do you use it?

**Comment by RadGH** [comment]
@GitHubRulesOK Thanks for clarifying and sorry for the delay. I had the need to try this again and just tested with your instructions. Looks like that dev build does get the fonts just fine :)

I wonder why this isn't part of the latest version? Hmmm!

![image](https://user-images.githubusercontent.com/2008464/45249433-1aab1d00-b2d5-11e8-9ae3-94278b4fd070.png)

**Comment by GitHubRulesOK** [comment]
Although you cant check each text element for font type, a bonus is you can highlight in colour (though NOT comment) any text you want others to change. Anything more detailed will require a pdf analyser.

**Comment by GitHubRulesOK** [comment]
@kjk since I have seen this feature working in alternative builds, is there any reason it cant be enabled in pre-release or better yet in full release ?

**Comment by GitHubRulesOK** [comment]
Closing as incorporated as part of wider feature request to more importantly highlight font failures per #350

**Comment by sahwar** [comment]
@kjk @ @GitHubRulesOK , you guys are amazing!
Great wrok!!!
